test/Microsoft.IdentityModel.Tests/ExpectedException.cs

```diff
@@ -143,15 +143,15 @@ public void ProcessException(Exception exception, List<string> errors = null)
                     return;
                 }
 
-                if (exception.InnerException != null && InnerTypeExpected == null)
+                if (exception.InnerException != null && InnerTypeExpected == null && !IgnoreInnerException)
                 {
-                    HandleError("exception.InnerException != null && expectedException.InnerTypeExpected == null.\nexception.InnerException: " + exception.InnerException, errors);
+                    HandleError("exception.InnerException != null && expectedException.InnerTypeExpected == null && IgnoreInnerException == false.\nexception.InnerException: " + exception.InnerException, errors);
                     return;
                 }
 
-                if (exception.InnerException == null && InnerTypeExpected != null)
+                if (exception.InnerException == null && InnerTypeExpected != null && !IgnoreInnerException)
                 {
-                    HandleError("exception.InnerException == null, expectedException.InnerTypeExpected != null.\nexpectedException.InnerTypeExpected: " + InnerTypeExpected, errors);
+                    HandleError("exception.InnerException == null && expectedException.InnerTypeExpected != null && IgnoreInnerException == false.\nexpectedException.InnerTypeExpected: " + InnerTypeExpected, errors);
```

Comment:
>IgnoreInnerException == false [](start = 116, length = 29)

why this construct vs ! ... the ! is used in most places in the code. Seems inconsistent to me (and not needed)